### PR TITLE
WIP: Updated wallet connect

### PR DESCRIPTION
**Pull request by baxri**
## Description

Wallet Connect - Update Web3Modal library for wallet signers

Task: [APP-2004](https://aragonassociation.atlassian.net/browse/APP-2004)

## Type of change

- [x] Update libraries

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2004]: https://aragonassociation.atlassian.net/browse/APP-2004?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/web-app/src/hooks/useWallet.tsx [hunk 1]
```diff
@@ -1,69 +1,98 @@
...
+    refreshChainId: async () => {
+      await new Promise(resolve => resolve(true));
+    },
+    disconnect: async () => {
+      await new Promise(resolve => {
+        disconnect();
+        resolve(true);
+      });
+    },
+  };
```
Comment (jamesej) [inline packages/web-app/src/hooks/useWallet.tsx:81]: Nothing wrong with this, just a tip: you can use Promise.resolve(value) to make an already resolved promise with the value passed in. So you could write
```
disconnect: () => {
  disconnect();
  return Promise.resolve(true);
}
```

File: packages/web-app/src/utils/constants/api.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ export const INFURA_PROJECT_ID: Record<
 };
 
 export const ARAGON_RPC = 'mainnet.eth.aragon.network';
+export const walletConnectProjectID = 'a405a1eafd0773a279ad607c01a546ae';
```
Comment (sepehr2github) [inline packages/web-app/src/utils/constants/api.ts]: Good practice: This needs to be handled in env file

File: packages/web-app/src/index.tsx [hunk 3]
```diff
@@ -19,27 +15,48 @@ import {TransactionDetailProvider} from 'context/transactionDetail';
...
-    options: {
-      infuraId: ARAGON_RPC,
-    },
-  },
-};
+const chains = [mainnet, goerli];
+
+// Wagmi client
+const {provider} = configureChains(chains, [
+  walletConnectProvider({projectId: walletConnectProjectID}),
```
Comment (sepehr2github) [inline packages/web-app/src/index.tsx:37]: As a provider we need to have Aragon RPC node to increase stability, please add aragon RPC here as a backup solution

File: packages/web-app/src/components/fullScreenStepper/fullScreenStepper.tsx [hunk 4]
```diff
@@ -105,6 +106,7 @@ export const FullScreenStepper: React.FC<FullScreenStepperProps> = ({
               description={wizardDescription || ''}
               totalSteps={totalSteps}
               currentStep={currentFormStep}
+              //@ts-ignore
```
Comment (sepehr2github) [inline packages/web-app/src/components/fullScreenStepper/fullScreenStepper.tsx]: Typescript is the only thing that protects us from many issues that might be existed especially in this project which has almost 0 test coverage, So please when you see a typescript issue please check the source and then disable it if there aren't any options, regardless this is redundant and should be removed, you just had to build the UI Component to update type interfaces :))

File: packages/web-app/src/pages/dashboard.tsx [hunk 5]
```diff
@@ -275,6 +276,7 @@ const Dashboard: React.FC = () => {
       <HeaderWrapper>
         <HeaderDao
           daoName={dao.metadata.name}
+          // @ts-ignore
```
Comment (sepehr2github) [inline packages/web-app/src/pages/dashboard.tsx]: Same as the pervious one should be removed, you just need to build the UI Components

File: packages/web-app/src/components/fullScreenStepper/fullScreenStepper.tsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
Comment (sepehr2github) [inline packages/web-app/src/components/fullScreenStepper/fullScreenStepper.tsx]: Also this one is not needed

File: packages/web-app/src/hooks/useWallet.tsx [hunk 7]
```diff
@@ -1,69 +1,98 @@
...
+      await new Promise(resolve => {
+        openWeb3Modal();
+        resolve({
+          networkId,
+          cacheProvider,
+        });
+      });
+    },
+    refreshChainId: async () => {
+      await new Promise(resolve => resolve(true));
```
Comment (sepehr2github) [inline packages/web-app/src/hooks/useWallet.tsx]: Could you explain a bit regarding this method?

**Review by sepehr2github — CHANGES_REQUESTED** [review]
Thanks, @baxri, This PR could solve many issues that we've experienced for wallet connection.

File: packages/web-app/src/pages/dashboard.tsx [hunk 8]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
Comment (sepehr2github) [inline packages/web-app/src/pages/dashboard.tsx]: Same

File: packages/web-app/src/index.tsx [hunk 9]
```diff
@@ -59,13 +84,17 @@ ReactDOM.render(
...
+                </WagmiConfig>
+              </NetworkProvider>
+            </AlertProvider>
+          </Router>
+        </APMProvider>
+      </PrivacyContextProvider>
+    </React.StrictMode>
+    <Web3Modal
+      projectId={walletConnectProjectID}
+      ethereumClient={ethereumClient}
```
Comment (sepehr2github) [inline packages/web-app/src/index.tsx:96]:
```suggestion
      ethereumClient={ethereumClient}
      themeMode="light"
```

Please add theme here to enable the light theme for web3 modal

**Review by sepehr2github — APPROVED** [review]
Thanks, one more change needed and then good to go

**Pull request merged**